Separate extension install from pip install

lbustelo opened the pull request:
fixes #165

Commit:
Modifications for new installation process
(c) Copyright IBM Corp. 2016

lbustelo commented:
Still pending changes to README

lbustelo commented:
All changes should be in... ready to merge

/cc @parente

parente commented:
Code looks like the right stuff. @nitind while you're looking  at jupyter-incubator/dashboards#166 can you give this one a shot too?

We're going to have to land these in a bit of a sequence. For example, the dashboards `make dev-with-widgets` currently assumes the old activate on pip install behavior and needs to update. I think a reasonably correct order is:

* declarativewidgets
* dashboards (needs update for decl widgets change in makefile)
* contentmanagement
* dashboards_bundlers (needs update for cms and dashboards changes)
* kernel_gateway_bundlers (needs update for cms changes)

Let's sit together and do the merges. Maybe do a release soon too?

lbustelo commented:
@parente @nitind I'm assuming no news is good news. Did everything check right with the widgets? I'm asking @peller to give this a once over from the development side. Make sure that `make dev` and `make test*` are working fine for him.

parente commented:
I'm giving it a once over again right now with dashboards too.

parente commented:
Did a `make sdist` in widgets. Updated the `make dev-with-widgets` in dashboards to do the install / activate commands. Saw this in the log:

```
[ExtensionActivateApp] Activating jupyter_declarativewidgets notebook server extensions
[ExtensionActivateApp] Activating jupyter_declarativewidgets JS notebook extensions
```

Saw this when running one of our most complex dashboard demos:

![screen shot 2016-01-20 at 10 11 27 am](https://cloud.githubusercontent.com/assets/153745/12452786/37081dda-bf5e-11e5-8208-dfc1e96d5617.png)

All good, except for the last commit re: Windows entry_points that should be ported here.

Commit:
Use setuptools entry_points when available
Properly installs top-level scripts as exes on Windows

(c) Copyright IBM Corp. 2016

peller commented:
make dev/test  - LGTM
